runtime-sdk/src/types/transaction.rs

```diff
@@ -116,12 +128,42 @@ pub struct Fee {
     pub gas: u64,
 }
 
+/// Common information that specifies an address as well as how to authenticate.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub enum AddressSpec {
+    #[serde(rename = "solo")]
+    Solo(PublicKey),
```

Comment:
Maybe `Single`? or `Singlesig`? No strong opinions though.

Reply:
forgot to mention this: in oasis-core, we proposed to not to have multiple structures allowed. you would instead make a multisig config with one key. we could do that here if we wanted. or maybe having a system with room for alternate ways to authenticate a transaction is a flexibility we like. or some combination of both where we choose keep an AddressSpec enum with only a multisig variant for now.

Reply:
thanks for the ideas. how about `Signature`/`Multisig`? there may be things other than multisig that "single" will be distinguished from in the future

Reply:
Sounds good to me!

Reply:
changed!